### PR TITLE
Added Unit Test For UserIdUtils

### LINKED ISSUE #3279: Write a Unit Test For UserIdUtils
Write a Unit Test for UserIdUtils

### PR DESCRIPTION
**Pull request by Smarshal21**
fixes #3279 

**Added Unit Test For UserIdUtils**
### 🏁 Checklist

- [x] ⛑️ Tests (unit and/or integration) are included 
- [x] 🔖 Capability is checked or not needed 
- [x] 🔙 Backport requests are created or not needed: `/backport to stable-xx.x`
- [x] 📅 Milestone is set
- [x] 🌸 PR title is meaningful (if it should be in the changelog: is it meaningful to users?)

### CODE REVIEW

**Comment by Smarshal21** [comment]
@mahibi @AndyScherzinger  as we had a chat i have started the writing tests by writing a simple unit test
could you ptal and let me know

File: app/build.gradle [hunk 1]
```diff
@@ -244,7 +244,9 @@ dependencies {
     implementation 'com.github.wooplr:Spotlight:1.3'
     implementation 'com.google.code.findbugs:jsr305:3.0.2'
     implementation 'com.github.nextcloud-deps:ChatKit:0.4.2'
-
+    testImplementation 'junit:junit:4.13.2'
```
Comment (AndyScherzinger) [inline app/build.gradle:247]: `junit` and `mockito` are duplicated, see https://github.com/nextcloud/talk-android/pull/3280/files#diff-51a0b488f963eb0be6c6599bf5df497313877cf5bdff3950807373912ac1cdc9R290 for the original entries and please then move the `mockito-android` dependency down there, so the test dependencies are grouped together.

File: app/src/test/java/com/nextcloud/talk/utils/UserIdUtilsTest.kt [hunk 2]
```diff
@@ -0,0 +1,36 @@
+package com.nextcloud.talk.utils
```
Comment (AndyScherzinger) [inline app/src/test/java/com/nextcloud/talk/utils/UserIdUtilsTest.kt:1]: Please add the usual license header as a comment, see for example https://github.com/nextcloud/talk-android/blob/master/app/src/main/java/com/nextcloud/talk/activities/BaseActivity.kt#L1-L21

**Review by AndyScherzinger — CHANGES_REQUESTED** [review]

**Comment by AndyScherzinger** [comment]
@Smarshal21 I added some minor comments. On top of that it would be nice if you could add a Github action to execute the tests. For inspiartion see https://github.com/nextcloud/android/blob/master/.github/workflows/unit-tests.yml while that would need to be rewriten since you won't have jacoco and the comment deletion. So a simple execution should be enough, like a modified version of https://github.com/nextcloud/android/blob/master/.github/workflows/unit-tests.yml#L32-L35 with a different, fitting value for https://github.com/nextcloud/android/blob/master/.github/workflows/unit-tests.yml#L35

**Comment by AndyScherzinger** [comment]
Also @Smarshal21 since we added you to the organization you can just create the branches within this repository. This makes your life and ours easier since rebases are a bit simpler and all GH actions will get executes since that is blocked on forks for security reasons.

**Comment by Smarshal21** [comment]
> Also @Smarshal21 since we added you to the organization you can just create the branches within this repository. This makes your life and ours easier since rebases are a bit simpler and all GH actions will get executes since that is blocked on forks for security reasons.

Sure

**Comment by Smarshal21** [comment]
> > @Smarshal21 I added some minor comments. On top of that it would be nice if you could add a Github action to execute the tests. For inspiartion see https://github.com/nextcloud/android/blob/master/.github/workflows/unit-tests.yml while that would need to be rewriten since you won't have jacoco and the comment deletion. So a simple execution should be enough, like a modified version of https://github.com/nextcloud/android/blob/master/.github/workflows/unit-tests.yml#L32-L35 with a different, fitting value for https://github.com/nextcloud/android/blob/master/.github/workflows/unit-tests.yml#L35
> 
> Yeah Thanks!! Will resolve them

Done

**Comment by Smarshal21** [comment]
Done

**Comment by Smarshal21** [comment]
> @Smarshal21 I added some minor comments. On top of that it would be nice if you could add a Github action to execute the tests. For inspiartion see https://github.com/nextcloud/android/blob/master/.github/workflows/unit-tests.yml while that would need to be rewriten since you won't have jacoco and the comment deletion. So a simple execution should be enough, like a modified version of https://github.com/nextcloud/android/blob/master/.github/workflows/unit-tests.yml#L32-L35 with a different, fitting value for https://github.com/nextcloud/android/blob/master/.github/workflows/unit-tests.yml#L35

could you tell me more about the github actions??

**Comment by AndyScherzinger** [comment]
> could you tell me more about the github actions??

@Smarshal21 anything specific besides the referenced workflow files?

**Comment by AndyScherzinger** [comment]
Sorry for the late reply, very busy these days :/

**Comment by Smarshal21** [comment]
> > could you tell me more about the github actions??
> 
> @Smarshal21 anything specific besides the referenced workflow files?

i wanna clarify about what else i have to include in this pr so that i can get an idea to write other tests as well

**Comment by Smarshal21** [comment]
> Sorry for the late reply, very busy these days :/

Thats fine :)

**Comment by mahibi** [comment]
i guess there was some confusion about the suggested github workflow.

The test itself is absolutely fine @Smarshal21 so i merged it.

Adding the github workflow is unrelated. It is about executing all tests on github, see https://docs.github.com/en/actions/using-workflows/triggering-a-workflow for explanation.
i am also not very familiar with it yet so i don't know off the top of my head what exactly is needed from https://github.com/nextcloud/android/blob/master/.github/workflows/unit-tests.yml
e.g. what has to be changed in uploadReport.sh etc..

**Pull request merged**